### PR TITLE
devops: check whether semgrep pro compiles with semgrep in PR

### PR DESCRIPTION
**Pull request by aryx**
test plan:
wait for green CI check

### CODE REVIEW

**Review by mjambon — APPROVED** [review]
Nice! You can now
* remove the checkbox from the PR `.github/workflows/pr-checklist.yaml`
* remove the xxx introduced to cause an error

File: src/core_scan/Core_scan.ml [hunk 1]
```diff
@@ -633,7 +633,7 @@ let iter_targets_and_get_matches_and_exn_to_errors config
 (* File targeting and rule filtering *)
 (*****************************************************************************)
 
-let xtarget_of_file ~parsing_cache_dir (xlang : Xlang.t) (file : Fpath.t) :
+let xxxtarget_of_file ~parsing_cache_dir (xlang : Xlang.t) (file : Fpath.t) :
```
Comment (ajbt200128) [inline src/core_scan/Core_scan.ml]: Why this change?

Comment (aryx) [reply src/core_scan/Core_scan.ml]: it was to test this PR, that is to test whether the new workflow would report an error because semgrep-pro relies on xtarget_of_file, so renaming this func in semgrep will break semgrep-pro

Comment (mjambon) [reply src/core_scan/Core_scan.ml]: I think he introduced an incompatibility on purpose to make sure the CI check would fail and block the merge.
